java/core/src/main/java/sleeper/core/util/LoggedDuration.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.core.util;
+
+import java.text.DecimalFormat;
+import java.time.Duration;
+import java.time.Instant;
+
+public class LoggedDuration {
+    private static final DecimalFormat FORMATTER = new DecimalFormat("0.###");
+    private final Duration duration;
+
+    private LoggedDuration(Duration duration) {
+        this.duration = duration;
+    }
+
+    public static LoggedDuration between(Instant start, Instant end) {
+        return new LoggedDuration(Duration.between(start, end));
+    }
+
+    public long getSeconds() {
+        return duration.getSeconds();
+    }
+
+    @Override
+    public String toString() {
+        return FORMATTER.format(duration.getSeconds() + (double) duration.getNano() / 1_000_000_000);
```

Comment:
We could avoid needing to add " seconds" to every log message by adding it here. It would also be good if this could support longer durations, eg. "2 minutes 42 seconds".

I also noticed there are cases when a short version is wanted, eg. in ArrayListRecordBatch. We could make that a boolean field for whether to output the short format?

Reply:
Added a `toShortString()` method which returns a shorthand string including hours, minutes and seconds. e.g `12h34m56.789s`